sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2ScanRelationPushDown.scala

```diff
@@ -17,61 +17,183 @@
+                    aggNode // return original plan node
+                  } else {
+                    // use the aggregate columns as the output columns
+                    // e.g. TABLE t (c1 INT, c2 INT, c3 INT)
+                    // SELECT min(c1), max(c1) FROM t;
+                    // Use min(c1), max(c1) as output for DataSourceV2ScanRelation
+                    // We want to have the following logical plan:
+                    // == Optimized Logical Plan ==
+                    // Aggregate [min(min(c1)#21) AS min(c1)#17, max(max(c1)#22) AS max(c1)#18]
+                    // +- RelationV2[min(c1)#21, max(c1)#22] parquet file ...
+                    val output = aggregates.map {
+                      case agg: AggregateExpression =>
+                        AttributeReference(toPrettySQL(agg), agg.dataType)()
+                    }
+
+                    // No need to do column pruning because only the aggregate columns are used as
+                    // DataSourceV2ScanRelation output columns. All the other columns are not
+                    // included in the output. Since PushDownUtils.pruneColumns is not called,
+                    // ScanBuilder.requiredSchema is not pruned, but ScanBuilder.requiredSchema is
+                    // not used anyways. The schema for aggregate columns will be built in Scan.
+                    val scan = sHolder.builder.build()
+
+                    logInfo(
+                      s"""
+                         |Pushing operators to ${sHolder.relation.name}
+                         |Pushed Aggregate Functions:
+                         | ${pushedAggregates.aggregateExpressions.mkString(", ")}
+                         |Output: ${output.mkString(", ")}
+                      """.stripMargin)
+
+                    val scanRelation = DataSourceV2ScanRelation(sHolder.relation, scan, output)
+                    val plan = Aggregate(groupingExpressions, resultExpressions, scanRelation)
+
+                    // Change the optimized logical plan to reflect the pushed down aggregate
+                    // e.g. TABLE t (c1 INT, c2 INT, c3 INT)
+                    // SELECT min(c1), max(c1) FROM t;
+                    // The original logical plan is
+                    // Aggregate [min(c1#9) AS min(c1)#17, max(c1#9) AS max(c1)#18]
+                    // +- RelationV2[c1#9] parquet ...
+                    //
+                    // After change the V2ScanRelation output to [min(_1)#21, max(_1)#22]
+                    // we have the following
+                    // !Aggregate [min(_1#9) AS min(_1)#17, max(_1#9) AS max(_1)#18]
+                    // +- RelationV2[min(_1)#21, max(_1)#22] parquet ...
+                    //
+                    // We want to change it to
+                    // == Optimized Logical Plan ==
+                    // Aggregate [min(min(c1)#21) AS min(c1)#17, max(max(c1)#22) AS max(c1)#18]
```

Comment:
why do we still have the `Aggregate`? grouping and aggregating are all done by the source already.

Reply:
If the aggregate doesn't have a group by, do we still need the `Aggregate` in Spark?

Reply:
I don't think so. The v2 source should respect the "no grouping key" semantic and only return one row.

Reply:
Sorry, I got confused here. Suppose we have a query `SELECT Max(col) FROM t`. We push down Max(col) to parquet, then for each partition file, we read footer to get Max(col), and these are the partial aggregate results. We still need to do a final aggregate, so I guess we still need the aggregate on RelationV2?